Packs/Accessdata/Integrations/AccessdataV2/AccessdataV2.py

```diff
@@ -0,0 +1,361 @@
+"""
+
+"""
+
+# python 3.9 imports
+from functools import wraps
+from json import dumps, loads
+from traceback import format_exc
+
+# accessdata imports
+from accessdata.api.agents import Agent
+from accessdata.api.jobs import Job
+from accessdata.api.filters import and_, or_
+from accessdata.client import Client
+
+# xsoar imports
+from CommonServerPython import *
+import demistomock as demisto
+
+""" decorator wrapping demisto commands """
+
+_run_functions = {}
+
+
+def wrap_demisto_command(command):
+    def _func(func):
+        @wraps(func)
+        def _inside(*args, **kwargs):
+            return func(*args, **kwargs)
+
+        _run_functions[command] = func
+        return _inside
+
+    return _func
+
+
+""" register demisto commands """
+
+
+@wrap_demisto_command("accessdata-api-get-case-by-name")
+def _get_case_by_name(client, name):
+    case = client.cases.first_matching_attribute("name", name)
+    if not case:
+        raise ValueError(f"Failed to gather case with name ({name}).")
+
+    return CommandResults(
+        outputs_prefix="Accessdata.Case",
+        outputs={
+            "ID": case["id"],
+            "Name": name,
+            "CaseFolder": case["ftkcasefolderpath"]
+        },
+        outputs_key_field="ID",
+        readable_output=tableToMarkdown("Case", dict(case))
+    )
+
+
+@wrap_demisto_command("accessdata-api-create-case")
+def _create_case(client, **kwargs):
+    case = client.cases.create(**kwargs)
+    if not case:
+        raise ValueError("Failed to create case.")
+
+    return CommandResults(
+        outputs_prefix="Accessdata.Case",
+        outputs={
+            "ID": case["id"],
+            "Name": case["name"],
+            "CaseFolder": case["ftkcasefolderpath"]
+        },
+        outputs_key_field="ID",
+        readable_output=tableToMarkdown("Case", dict(case))
+    )
+
+
+@wrap_demisto_command("accessdata-api-process-evidence")
+def _process_evidence(client, caseid, evidence_path, evidence_type, options):
+    # gather the case object from it's id
+    case = client.cases.first_matching_attribute("id", int(caseid))
+    if not case:
+        raise ValueError(f"Failed to gather case with id ({caseid}).")
+
+    # try find json in the options
+    try:
+        options = loads(options)
+    except Exception:
```

Comment:
I need to error on anything here rather than just the decode error as the resultant object is passed into the SDK. I can build a large list of exceptions but think it is simpler to leave as broad term for this one.